core/services/ocr2/plugins/s4/plugin.go

```diff
@@ -27,7 +27,7 @@ type key struct {
 
 var _ types.ReportingPlugin = (*plugin)(nil)
 
-func NewReportingPlugin(logger logger.Logger, config *PluginConfig, orm s4.ORM) (types.ReportingPlugin, error) {
+func NewReportingPlugin(logger commontypes.Logger, config *PluginConfig, orm s4.ORM) (types.ReportingPlugin, error) {
```

Comment:
I'd suggest to rename this method to not collide with the OCR Factory interface one. Or maybe even move those validations into the factory method and get rid of this one completely?

Reply:
This is convenient for tests to have this constructor function. I can rename it to NewReportingPluginInstance for example, in the next PR.